src/profile-logic/import/instruments/instruments.js

```diff
@@ -0,0 +1,746 @@
+  getEmptyThread,
+} from '../../../profile-logic/data-structures';
+import BinaryPlistParser, { UID } from './BinaryPlistParser';
+import BinReader from './BinReader';
+
+// TODO make helpers.js and move the appropriate helper functions into it
+// TODO add the missing return types in the functions
+// TODO I have used console.log() statements for debug purpose, I will remove them later
+// TODO add Root frame at the top
+// TODO fill remaining meta data inside profile
+
+let fileReader;
+
+function parseBinaryPlist(bytes) {
+  const text = 'bplist00';
+  for (let i = 0; i < 8; i++) {
+    if (bytes[i] !== text.charCodeAt(i)) {
+      throw new Error('File is not a binary plist');
+    }
+  }
+
+  return new BinaryPlistParser(
+    new DataView(bytes.buffer, bytes.byteOffset, bytes.byteLength)
+  ).parseRoot();
+}
+
+function decodeUTF8(bytes: Uint8Array): string {
+  let text = String.fromCharCode.apply(String, bytes); // eslint-disable-line prefer-spread
+  if (text.slice(-1) === '\0') text = text.slice(0, -1); // Remove a single trailing null character if present
+  return decodeURIComponent(escape(text));
+}
+
+function followUID(objects: any[], value: any): any {
+  return value instanceof UID ? objects[value.index] : value;
+}
+
+function sortBy<T>(ts: T[], key: (t: T) => number | string): void {
+  function comparator(a: T, b: T) {
+    return key(a) < key(b) ? -1 : 1;
+  }
+  ts.sort(comparator);
+}
+
+function patternMatchObjectiveC(
+  objects: any[],
+  value: any,
+  interpretClass: ($classname: string, obj: any) => any = x => x
+): any {
+  if (isDictionary(value) && value.$class) {
+    const name = followUID(objects, value.$class).$classname;
+    switch (name) {
+      case 'NSDecimalNumberPlaceholder': {
+        const length: number = value['NS.length'];
+        const exponent: number = value['NS.exponent'];
+        const byteOrder: number = value['NS.mantissa.bo'];
+        const negative: boolean = value['NS.negative'];
+        const mantissa = new Uint16Array(
+          new Uint8Array(value['NS.mantissa']).buffer
+        );
+        let decimal = 0;
+
+        for (let i = 0; i < length; i++) {
+          let digit = mantissa[i];
+
+          if (byteOrder !== 1) {
+            digit = ((digit & 0xff00) >> 8) | ((digit & 0x00ff) << 8);
+          }
+
+          decimal += digit * Math.pow(65536, i);
+        }
+
+        decimal *= Math.pow(10, exponent);
+        return negative ? -decimal : decimal;
+      }
+
+      // Replace NSData with a Uint8Array
+      case 'NSData':
+      case 'NSMutableData':
+        return value['NS.bytes'] || value['NS.data'];
+
+      // Replace NSString with a string
+      case 'NSString':
+      case 'NSMutableString':
+        return decodeUTF8(value['NS.bytes']);
+
+      // Replace NSArray with an Array
+      case 'NSArray':
+      case 'NSMutableArray':
+        if ('NS.objects' in value) {
+          return value['NS.objects'];
+        }
+        const array: any[] = [];
```

Comment:
You are making a lexical declaration here but it doesn't have a block to bound to. You should create a block for them like this: 
```js
case 'something': {
  // const/let declaration
  // ...
  break;
}
```
(some for some of the case statements below)